src/data/mv_polynomial/basic.lean

```diff
@@ -90,7 +90,7 @@ variables {R : Type u} {S₁ : Type v} {S₂ : Type w} {S₃ : Type x}
 
 /-- Multivariate polynomial, where `σ` is the index set of the variables and
   `R` is the coefficient ring -/
-def mv_polynomial (σ : Type*) (R : Type*) [comm_semiring R] := add_monoid_algebra R (σ →₀ ℕ)
+def mv_polynomial (σ : Type*) (R : Type*) [semiring R] := add_monoid_algebra R (σ →₀ ℕ)
```

Comment:
wow, this is really surprising that it wasn't already here. I think this need some documentation tweaking; was this a previous deliberate decisiion?

Reply:
It's not so surprising to me since (this definition of) `mv_polynomial` doesn't make sense viewed as "polynomial function" unless it's a `comm_semiring`. Namely, we have `X 0 * X 1 = X 1 * X 0` also if multiplication is not commutative. Still if we can generalize for free, I don't see why not.

Reply:
We tried this a while ago in #6481, but it was just too hard to fix everything dowstream so it rotted.